Client/game_sa/CDummyPoolSA.cpp

```diff
@@ -13,63 +13,93 @@
 
 #include "StdInc.h"
 #include "CDummyPoolSA.h"
+#include "CGameSA.h"
+#include <game/CWorld.h>
+
+extern CGameSA* pGame;
 
 CDummyPoolSA::CDummyPoolSA()
 {
     m_ppDummyPoolInterface = (CPoolSAInterface<CEntitySAInterface>**)0xB744A0;
 }
 
-void CDummyPoolSA::RemoveAllBuildingLods()
+void CDummyPoolSA::RemoveAllWithBackup()
 {
-    if (m_pLodBackup)
+    if (m_pOriginalElementsBackup)
         return;
 
-    m_pLodBackup = std::make_unique<std::array<CEntitySAInterface*, MAX_DUMMIES>>();
+    m_pOriginalElementsBackup = std::make_unique<pool_backup_t>();
 
-    for (int i = 0; i < MAX_DUMMIES; i++)
+    auto pDummyPool = (*m_ppDummyPoolInterface);
+    for (size_t i = 0; i < MAX_DUMMIES_DEFAULT; i++)
     {
-        CEntitySAInterface* object = (*m_ppDummyPoolInterface)->GetObject(i);
-        (*m_pLodBackup)[i] = object->m_pLod;
-        object->m_pLod = nullptr;
+        if (pDummyPool->IsContains(i))
+        {
+            CEntitySAInterface* building = pDummyPool->GetObject(i);
+
+            pGame->GetWorld()->Remove(building, CDummyPool_Destructor);
+            building->RemoveRWObjectWithReferencesCleanup();
+
+            pDummyPool->Release(i);
+
+            (*m_pOriginalElementsBackup)[i].first = true;
+            (*m_pOriginalElementsBackup)[i].second = *building;
+        }
+        else
+        {
+            (*m_pOriginalElementsBackup)[i].first = false;
+        }
```

Comment:
Early return (continue)

Reply:
early continue is the same as early return

Reply:
No, 'return' finishes a function. 'continue' is syntax sugar for 'goto' to next loop iteration

Reply:
> No, 'return' finishes a function. 'continue' is syntax sugar for 'goto' to next loop iteration

Yes, but the idea stays the same.

Reply:
No, they have different readability. Avoid 'continue' where it's unnecessary

Reply:
> No, they have different readability. Avoid 'continue' where it's unnecessary

absolutely not.
`return` skips the content of the block in a funciton.
`continue` skips the content of the block in a codeblock in loop

Reply:
No, you're overusing it. I see no reason to put `continue` in every loop where it doesn't improve readability.

Reply:
It absolutely does improve readability